SteamWorkshopService/KeyValueExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NLog;
+using SteamKit2;
+
+namespace SteamWorkshopService
+{
+    public static class KeyValueExtensions
+    {
+        private static Logger Log = LogManager.GetLogger("SteamWorkshopService");
+
+        public static T GetValueOrDefault<T>(this KeyValue kv, string key)
+        {
+            kv.TryGetValueOrDefault<T>(key, out T result);
+            return result;
+        }
+        public static bool TryGetValueOrDefault<T>(this KeyValue kv, string key, out T typedValue)
+        {
+            var match = kv.Children?.Find((KeyValue item) => item.Name == key);
+            object result = default(T);
+            if (match == null)
+            {
+                typedValue = (T) result;
+                return false;
+            }
+
+            var value = match.Value ?? "";
+            var success = false;
+
+            if (typeof(T) == typeof(UInt64))
```

Comment:
See string to value conversion done in the command system, [here](https://github.com/TorchAPI/Torch/blob/master/Torch/Commands/Command.cs#L144-L158)

Reply:
Ugh. I would have needed you back when I wrote that method :D